src/main/java/spoon/MavenLauncher.java

```diff
@@ -109,657 +120,99 @@ public MavenLauncher(String mavenProject, String m2RepositoryPath, SOURCE_TYPE s
-				RangeVersion rangeVersion = new RangeVersion(version);
-				for (int i = 0; i < versionsFromM2.size(); i++) {
-					Version v = versionsFromM2.get(i);
-					if (rangeVersion.include(v)) {
-						version = v.version;
-						break;
-					}
-				}
-			}
-			return version;
-		}
-
-		private InheritanceModel readPom(String groupId, String artifactId, String version) {
-			version = extractVersion(groupId, artifactId, version);
-			groupId = groupId.replace(".", "/");
-			String fileName = artifactId + "-" + version;
-			Path depPath = Paths.get(m2RepositoryPath, groupId, artifactId, version, fileName + ".pom");
-			try {
-				return readPOM(depPath.toString(), null);
-			} catch (Exception e) {
-				LOGGER.log(Level.ERROR, "Unable to read the POM:" + depPath.toString(), e);
-				return null;
-			}
 		}
 
-		private TreeDependency getTreeDependency(Dependency dependency, boolean isLib, Set<TreeDependency> hierarchy) {
-			String groupId = extractVariable(dependency.getGroupId());
-			String artifactId = extractVariable(dependency.getArtifactId());
-			String version = extractVersion(groupId, artifactId, dependency.getVersion());
-			if (version == null) {
-				LOGGER.warn("A dependency version cannot be resolved: " + groupId + ":" + artifactId + ":" + version);
-				return null;
-			}
-			// pass only the optional dependency if it's in a library dependency
-			if (isLib && dependency.isOptional()) {
-				return null;
-			}
-
-			// ignore test dependencies for app source code
-			if ("test".equals(dependency.getScope()) && (SOURCE_TYPE.APP_SOURCE == sourceType || isLib)) {
-				return null;
-			}
-			// ignore not transitive dependencies
-			if (isLib && ("test".equals(dependency.getScope()) || "provided".equals(dependency.getScope()) || "compile".equals(dependency.getScope()))) {
-				LOGGER.log(Level.WARN, "Dependency ignored (scope: provided or test):" + dependency.getGroupId() + ":" + dependency.getArtifactId() + ":" + version);
-				return null;
-			}
-			TreeDependency dependence = new TreeDependency(groupId, artifactId, version, dependency.getType());
-			try {
-				InheritanceModel dependencyModel = readPom(groupId, artifactId, version);
-				if (dependencyModel != null) {
-					dependence = dependencyModel.getTreeDependency(true, hierarchy);
-					dependence.groupId = groupId;
-					dependence.artifactId = artifactId;
-					dependence.version = version;
-
-					if (dependency.getExclusions() != null) {
-						for (int i = 0; i < dependency.getExclusions().size(); i++) {
-							Exclusion exclusion = dependency.getExclusions().get(i);
-							dependence.removeDependency(exclusion.getGroupId(), exclusion.getArtifactId());
-						}
-					}
-				}
+		return classpath;
+	}
 
-			} catch (Exception e) {
-				LOGGER.log(Level.ERROR, "Unable to read the pom of the dependency:" + dependence.toString(), e);
-			}
-			return dependence;
+	private File getDefaultMavenHome() throws SpoonException {
```

Comment:
outch! This method looks really ugly!
Can't you rely on environment variable instead? The `M2_HOME` was considered as a standard one, but it's not heavily used.

Reply:
Oh you saw that...Fine, I'll correct it.